client/app/lib/contacts_listener.coffee

```diff
@@ -25,15 +29,54 @@ module.exports = class ContactsListener extends CozySocketListener
                 options = {}
             else
                 action  = 'add'
-                options = merge: true
-
-        @collection[action] @queues[queue], options
-        @queues[queue].length = 0
+                options =
+                    merge: true
+                    trigger: false
+
+
+        if @enabled
+            console.log """
+            Remote action occured (#{queue}, #{action}), processing start!
+            """
+
+            @collection.disableSort()
+            @end = new Date().getTime()
+
+            # Build batches of 10 models on which to perform operations.
+            lists = []
+            nbOccurences = @queues[queue].length
+            while @queues[queue].length > 0
+                lists.push @queues[queue].splice 0, 10
```

Comment:
In fact we are not at this stage of performance improvement. We have a problem due to the contact list that is too big and requires many filtering and sorting operation. I don't know how to change it properly, so I added dirty hacks everywhere performance problems were too high.